File: Array/Minimum Swaps.go
package main

import "fmt"

func main() {

	A, B := []int{1, 12, 10, 3, 14, 10, 5}, 8
	result := minSwaps(A, B)
	fmt.Println(result)
}

func minSwaps(A []int, B int) int {
	minCount, ans := 0, 0

	for i := 0; i < len(A); i++ {
		if A[i] <= B {
			minCount++
		}
	}
	if minCount <= 1 {
		return 0
	} else {
		leftCount, rightCount, count := 0, 0, 0
		for rightCount < minCount {
			if A[rightCount] > B {
				count++
			}
			rightCount++
		}
		ans = count
		for rightCount < len(A) {
			if A[rightCount] > B {
				count++
			}
			if A[leftCount] > B {
				count--
			}
			if count < ans {
				ans = count
			}
			rightCount++
			leftCount++
		}
	}

	return ans
}

File: Pythons Solutions/Minimum Swaps.py
from turtle import right


class Solution:
    def minSwaps(self, arr, b):
        minCount = 0

        for i in range(len(arr)):
            if arr[i] < b:
                minCount +=1
        
        if minCount <= 1:
            return 0
        else:
            rightCount, leftCount, count = 0, 0, 0
            while rightCount < minCount:
                if arr[rightCount] > b:
                    count += 1
                rightCount += 1
            ans = count
            while rightCount < len(arr):
                if arr[rightCount] > b:
                    count += 1
                if arr[leftCount] > b:
                    count -= 1
                
                ans = min(ans, count)

                rightCount += 1
                leftCount += 1

        return ans

obj = Solution()
arr, b = [1, 12, 10, 3, 14, 10, 5], 8
ans = obj.minSwaps(arr, b)
print(ans)